contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,320 @@
+
+        require(playerTwoMove >= Moves.ROCK, INVALID_MOVE_MSG);
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_ONE_REVEALED, INVALID_STEP_MSG);
+        address playerTwo = game.playerTwo;
+
+        //Validate move
+        bytes32 expectedMoveHash = createPlayerTwoMoveHash(msg.sender, gameKey, secret, playerTwoMove);
+        require(game.playerTwoMoveHash == expectedMoveHash, "Move and secret do not match");
+
+        emit PlayerTwoReveals(gameKey, msg.sender, playerTwoMove);
+
+        uint256 stake = game.stake;
+        address playerOne = game.playerOne;
+        address winner;
+        address loser;
+        GameResult outcome = resolveGame(game.playerOneMove, playerTwoMove);
+
+        if (outcome == GameResult.DRAW) {
+            resetGame(game);
+            balances[playerOne] = balances[playerOne].add(stake);
+            balances[playerTwo] = balances[playerTwo].add(stake);
+            DrawRefund(gameKey, playerOne, playerTwo, stake, stake);
+
+            return;
+
+        } else if (outcome == GameResult.LEFTWIN) {
+            winner = playerOne;
+            loser = playerTwo;
+        } else if (outcome == GameResult.RIGHTWIN) {
+            winner = playerTwo;
+            loser = playerOne;
+        } else {
+            assert(false);
+        }
+        uint256 winnings = stake.mul(2);
+        balances[winner] = balances[winner].add(winnings);
+        emit GamePayment(gameKey, winner, winnings);
+
+        resetGame(game);
+    }
+
+    function resolveGame(Moves leftPlayer, Moves rightPlayer) public pure returns(GameResult) {
+
+        if (leftPlayer == Moves.EMPTY || rightPlayer == Moves.EMPTY){
+            return GameResult.INCORRECT;
+        }
+
+        return GameResult((uint8(leftPlayer).add(3).sub(uint8(rightPlayer))).mod(3));
+    }
+
+    function playerOneCollectsForfeit(bytes32 gameKey) external whenNotPaused {
+
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_ONE_REVEALED, INVALID_STEP_MSG);
+        address playerOne = game.playerOne;
+        require(msg.sender == playerOne, INVALID_PLAYER_MSG);
+        require(block.timestamp >= game.expiryDate, GAME_NOT_EXPIRED_MSG);
+
+        uint256 forfeit = game.stake.mul(2);
+        resetGame(game);
+        balances[playerOne] = balances[playerOne].add(forfeit);
+
+        emit ForfeitPaid(gameKey, playerOne, forfeit);
+    }
+
+    function playerTwoCollectsForfeit(bytes32 gameKey) external whenNotPaused {
+
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_TWO_MOVED, INVALID_STEP_MSG);
+        address playerTwo = game.playerTwo;
+        require(msg.sender == playerTwo, INVALID_PLAYER_MSG);
+        require(block.timestamp >= game.expiryDate, GAME_NOT_EXPIRED_MSG);
+
+        uint256 forfeit = game.stake.mul(2);
+        resetGame(game);
+        balances[playerTwo] = balances[playerTwo].add(forfeit);
```

Comment:
I am thinking that since you had `require(msg.sender == playerTwo`, you could in fact skip using a memory `playerTwo` and instead use `msg.sender` elsewhere.